theano/tensor/nnet/__init__.py

```diff
@@ -30,3 +30,91 @@
                    scalar_sigmoid, ultra_fast_sigmoid,
                    hard_sigmoid)
 from .bn import batch_normalization
+
+
+import warnings
+from .abstract_conv import conv2d as abstract_conv2d
+
+def conv2d(input, filters, input_shape=None, image_shape=None, filter_shape=None,
+           border_mode='valid', subsample=(1, 1), filter_flip=True,
```

Comment:
This is not good, the order of arguments should be as for `conv2d()`. I.e., `image_shape` should be added in the end or scraped out of `**kwargs`, not interspersed with the others. Otherwise, existing code using positional arguments (such as `conv2d(x, y, x_shape, y_shape)`) will fail.